src/classes/TriggerHandler.cls

```diff
@@ -1,240 +1,159 @@
 public virtual class TriggerHandler {
-
-  // static map of handlername, times run() was invoked
-  private static Map<String, LoopCount> loopCountMap;
-  private static Set<String> bypassedHandlers;
-
-  // the current context of the trigger, overridable in tests
-  @TestVisible
-  private TriggerContext context;
-
-  // the current context of the trigger, overridable in tests
-  @TestVisible
-  private Boolean isTriggerExecuting;
-
-  // static initialization
-  static {
-    loopCountMap = new Map<String, LoopCount>();
-    bypassedHandlers = new Set<String>();
-  }
-  
-  // constructor
-  public TriggerHandler() {
-    this.setTriggerContext();
-  }
-
-  /***************************************
-   * public instance methods
-   ***************************************/
-
-  // main method that will be called during execution
-  public void run() {
-
-    if(!validateRun()) {
-      return;
+    @TestVisible
+    private Boolean testMode = false;
+    private System.TriggerOperation triggerEvent;
+    private static Map<String, LoopCount> loopCountByTrigerHandlerName = new Map<String, LoopCount>();
+    private static Set<String> bypassedHandlers = new Set<String>();
+    
+    public TriggerHandler(System.TriggerOperation triggerEvent) {
```

Comment:
@jeesparza2 It looks like you're requiring passing in the trigger event each time - changing the constructor. This is a breaking change, so nobody can update code to use the latest version if your PR is merged and they copy/paste your code over existing code.

Perhaps you can add your constructor as another option but leave the old code - that way it's not a breaking change.
Please also update documentation. There is a readme file and some other documentation.

(Shameless plug: If you'd like to do that on my repo, I'd include your additional constructor in that. https://github.com/dschach/salesforce-trigger-framework)